lib/ex_subtil_backend/workflow/step/set_language.ex

```diff
@@ -69,31 +73,33 @@ defmodule ExSubtilBackend.Workflow.Step.SetLanguage do
     end
   end
 
-  defp get_audio_source_files(_jobs, _priority_job_names, result \\ [])
-  defp get_audio_source_files([], _priority_job_names, result), do: result
-  defp get_audio_source_files(_jobs, [], result), do: result
-  defp get_audio_source_files(jobs, priority_job_names, result) do
-
-    {jobs, priority_job_names, result} =
-      Enum.find(jobs, fn(job) -> job.name == List.first(priority_job_names) end)
-      |> case do
-          nil ->
-            {jobs, List.delete_at(priority_job_names, 0), result}
-          job ->
-            jobs = List.delete(jobs, job)
-
-            case get_job_destination_files(job) do
-              nil -> {jobs, List.delete_at(priority_job_names, 0), result}
-              job_dest_path ->
-                if Enum.find(result, fn(file) -> Path.basename(file) == Path.basename(job_dest_path) end) do
-                  {jobs, List.delete_at(priority_job_names, 0), result}
-                else
-                  {jobs, priority_job_names, List.insert_at(result, -1, job_dest_path)}
-                end
-            end
-        end
-
-    get_audio_source_files(jobs, priority_job_names, result)
+  defp get_audio_jobs(_jobs, result \\ [])
+  defp get_audio_jobs([], result), do: result
+  defp get_audio_jobs([job | jobs], result) do
```

Comment:
`get_related_jobs` is a better name (and prevent steps dependencies definition)

Reply:
Done in a072e75!